frontend/src/app/modules/settings/user/user-profile-page/user-profile-page.component.ts

```diff
@@ -111,20 +111,28 @@ export class UserProfilePageComponent extends BaseComponent implements OnInit {
                     this.changeCountryCode(this.userForm);
                 }
             });
+
+        this.settingPageService.updateButtonClickEvent$.pipe(this.untilThis).subscribe(() => {
+            this.OnSubmit();
+        });
+
+        this.userForm.statusChanges.pipe(this.untilThis).subscribe(() => {
+            this.settingPageService.updateButtonActive(!this.userForm.invalid && !this.userForm.pending && this.userForm.dirty);
+        });
     }
 
-    public OnSubmit(form: FormGroup) {
+    public OnSubmit() {
         if (this.user) {
+            this.userForm.reset(this.userForm.value);
```

Comment:
This `reset()` sets all form controls to default values, thus user, as payload for request, will always have null fields. 

We have `.pipe(finalize(() => this.userForm.markAsPristine()))`, which makes form `!dirty` and disables the update button. 

And that's still not quite right, because the form will be marked pristine, even if the request fails